### PR TITLE
Add warmup iteration option to cholesky and triangular solver miniapps

### PR DESCRIPTION
**Pull request by msimberg**

### CODE REVIEW

**Review by teonnik — APPROVED** [review]

**Comment by msimberg** [comment]
bors try

**Comment by rasolca** [comment]
What about adding it to the Cholesky miniapp too?

**Review by albestro — APPROVED** [review]
Nice idea, LGTM

As @rasolca said, I would add it to cholesky miniapp too.

**Comment by msimberg** [comment]
Good idea! I'll update this to add it to the cholesky miniapp as well.

File: miniapp/miniapp_cholesky.cpp [hunk 1]
```diff
@@ -392,13 +394,16 @@ options_t check_options(hpx::program_options::variables_map& vm) {
...
+      vm["nruns"].as<int64_t>(),        vm["nwarmups"].as<int64_t>(),
+      CHECK_RESULT::NONE,
   };
 
   DLAF_ASSERT(opts.m > 0, "matrix size must be a positive number!", opts.m);
   DLAF_ASSERT(opts.mb > 0, "block size must be a positive number!", opts.mb);
   DLAF_ASSERT(opts.grid_rows > 0, "number of grid rows must be a positive number!", opts.grid_rows);
   DLAF_ASSERT(opts.grid_cols > 0, "number of grid columns must be a positive number!", opts.grid_cols);
+  DLAF_ASSERT(opts.nruns > 0, "number of runs must be a positive number!", opts.nruns);
+  DLAF_ASSERT(opts.nwarmups >= 0, "number of warmup runs must be a nonnegative number!", opts.nwarmups);
```
Comment (msimberg) [inline miniapp/miniapp_cholesky.cpp:406]: I suppose the message will not stay for too long. Would you like me to leave removing it to #295 or get rid of it right away?

**Comment by msimberg** [comment]
bors try

**Comment by msimberg** [comment]
bors try-

**Comment by msimberg** [comment]
bors try

File: miniapp/miniapp_cholesky.cpp (hunk 1 above)
Comment (teonnik) [reply miniapp/miniapp_cholesky.cpp:405]: Yes, just go ahead and remove it, it's redundant.

Comment (msimberg) [reply miniapp/miniapp_cholesky.cpp]: Done.

**Review by rasolca — APPROVED** [review]

**Comment by rasolca** [comment]
bors try

**Pull request merged**